listener/kcp/workspace_config.go

```diff
@@ -45,7 +48,32 @@ func virtualWorkspaceConfigFromCfg(ctx context.Context, appCfg config.Config, re
 		return nil, ErrEmptyVirtualWorkspaceURL
 	}
 
-	restCfg.Host = virtualWorkspaceURL
+	internalVirtualWorkspaceURL, err := combineBaseURLAndPath(restCfg.Host, virtualWorkspaceURL)
+	if err != nil {
+		return nil, err
+	}
+
+	restCfg.Host = internalVirtualWorkspaceURL
 
 	return restCfg, nil
 }
+
+func combineBaseURLAndPath(baseURLStr, pathURLStr string) (string, error) {
+	baseURL, err := url.Parse(baseURLStr)
+	if err != nil {
+		return "", errors.Join(ErrInvalidURL, err)
+	}
+
+	pathURL, err := url.Parse(pathURLStr)
+	if err != nil {
+		return "", errors.Join(ErrInvalidURL, err)
+	}
+
+	path := pathURL.Path
+
+	if !strings.HasPrefix(path, "/") {
+		path = "/" + path
+	}
+
+	return baseURL.ResolveReference(&url.URL{Path: path}).String(), nil
```

Comment:
can't you just use url.JoinPath(baseURLStr, pathURL) here instead?

Reply:
No, it does simple concatenation, and instead of:
```
"https://openmfp-kcp-front-proxy.openmfp-system:8443/services/apiexport/root/kubernetes.graphql.gateway"
```
It end up with
```
"https://openmfp-kcp-front-proxy.openmfp-system:8443/clusters/root/https:/kcp.dev.local:8443/services/apiexport/root/kubernetes.graphql.gateway"
```
(host from second param is not removed)